src/Computaria/Tardis/Factory.php

```diff
@@ -21,7 +20,7 @@ public static function grow(
         $beforeMethodCall = new Interceptor\CacheFetch($cacheAdapter, $idGenerator);
         $afterMethodCall = new Interceptor\CacheSave($cacheAdapter, $idGenerator);
 
-        return new Proxy\PublicMethods($proxyFactory, $beforeMethodCall, $afterMethodCall);
+        return new Proxy\PublicMethods($proxyFactory, $beforeMethodCall, $afterMethodCall, $idGenerator, $cacheAdapter);
```

Comment:
Both of these arguments [should not be added here][1], they should be an `Interceptor` (note that they already receive both arguments because they need them).

[1]: https://en.wikipedia.org/wiki/Law_of_Demeter "Law of Demeter"